zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -130,12 +134,28 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
             [cd_address.address(), rd_address.address()],
         );
 
+        // `code_address` is poped from stack and used to check if it exists in access
+        // list and get code hash. Caller and callee addresses are `parent_address` and
+        // `current_address` for opcode `DELEGATECALL`, otherwise are `current_address`
+        // and `code_address`.
+        let code_address = from_bytes::expr(&code_address_word.cells[..N_BYTES_ACCOUNT_ADDRESS]);
+        let caller_address = select::expr(
+            is_delegatecall.expr(),
+            parent_address.expr(),
+            current_address.expr(),
+        );
+        let callee_address = select::expr(
+            is_delegatecall.expr(),
+            current_address.expr(),
+            code_address.expr(),
+        );
+
         // Add callee to access list
         let is_warm = cb.query_bool();
         let is_warm_prev = cb.query_bool();
         cb.account_access_list_write(
             tx_id.expr(),
-            callee_address.clone(),
+            code_address.expr(),
```

Comment:
suppose to be still `callee_address` as L146 handle it correctly if `delegatecall`

Reply:
Fixed to replace `code_address` with `callee_address`.